neptune/mockedneptune.go

```diff
@@ -0,0 +1,24 @@
+package neptune
+
+import (
+	"github.com/ONSdigital/dp-graph/neptune/internal"
+)
+
+/*
+This module provides the MockDB factory function to make a NeptuneDB into
+which a mocked implementation of the gremgo driver's Pool may be injected
+to avoid real database access.
+*/
+
+import (
+	"github.com/ONSdigital/dp-graph/neptune/driver"
+)
+
+// MockDB provides a NeptuneDB, into which you can pass a mocked
+// NeptunePoolMock implementation, and thus write tests that bypass real
+// database communication.
+func mockDB(poolMock *internal.NeptunePoolMock) *NeptuneDB {
```

Comment:
quite a small file, so I (personally) would be tempted to put this func into `mockpoolfns.go`

Reply:
mockDB needs to be in a different package from `internal` in order to avoid circular imports. Specifically the tests in `neptune` . depend on (for example) `internal.ReturnOne` to inject test-specific behaviour to the driver's `Pool`. They must therefore import `internal`. If mockDB was itself also in `internal` it would want to import `neptune` to reach `neptune.NeptuneDB`, and would fail because that creates the circular import path. We could put mockDB in a package of its own - but that does not address the original point - that it is a very small file to exist on its own. ?